### PR TITLE
Unify reconnection and message processing in mcp client.

### PR DESCRIPTION
**Pull request by ayj**
Merge the (re)connection and processing logic into a single function
and unify the error handling and retry code paths. Update the ctrlz
tests to account for the fact that the initial MCP requests are now
captured in the journel.

This also adds unit tests for the reconnection logic on send and
receive failures.

### CODE REVIEW

File: pkg/mcp/client/client.go [hunk 1]
```diff
@@ -247,138 +245,96 @@ func (c *Client) handleResponse(response *mcp.MeshConfigResponse) error {
...
-		select {
-		case response, more := <-responseC:
-			if !more {
-				log.Errorf("Stream receive error: %v", responseError)
-
-				// re-establish the stream and start a new receive goroutine
-				// on any kind of error until the context is canceled.
-				if err := c.openStream(ctx); err != nil {
-					return
+		// The version and nonce information is reset for each new
```
Comment (ozevren) [inline pkg/mcp/client/client.go]: It is not clear why we have to stage the initial requests. What is the root problem?

Comment (ayj) [reply pkg/mcp/client/client.go]: Updated the comments to hopefully clarify why we need to send initial requests.

**Comment by ozevren** [comment]
/lgtm
/approve
There are some linter/build issues. I can re-lgtm after thos are fixed.

**Pull request merged**